### PR TITLE
Read provenance tags from serialized graph 

### PR DESCRIPTION
**Pull request by ransford2011**
This enables the ability to read provenance tags from the serialized graph and ensure that the values are preserved even when the graph is cloned.

### CODE REVIEW

File: src/ngraph/graph_util.cpp [hunk 1]
```diff
@@ -233,6 +233,14 @@ std::list<std::shared_ptr<ngraph::Node>>
                 // There is a friendly name for this node so copy it
                 cloned_node->set_friendly_name(node->get_friendly_name());
             }
+
+            if (!node->get_provenance_tags().empty())
+            {
```
Comment (diyessi) [inline src/ngraph/graph_util.cpp]: No need to test for `empty` since the `for` is testing for it implicitly.

File: src/ngraph/serializer.cpp [hunk 2]
```diff
@@ -1783,6 +1783,14 @@ shared_ptr<Node> JSONDeserializer::deserialize_node(json& node_js)
         {
             node->set_friendly_name(node_name);
         }
+        std::vector<json> prov_js = node_js.at("provenance_tags");
+        if (!prov_js.empty())
+        {
```
Comment (diyessi) [inline src/ngraph/serializer.cpp]: Same comment about testing for empty.

**Comment by aslepko** [comment]
TESTNOW
